### PR TITLE
adds documentation for the configuration options in local run

### PR DESCRIPTION
**Pull request by maasg**
Adds documentation about the configuration options, syntax, and usage for the _sandbox_.

### CODE REVIEW

**Comment by maasg** [comment]
(should also merge to 2.0.5-docs)

File: docs/shared-content-source/docs/modules/develop/pages/cloudflow-local-sandbox.adoc [hunk 1]
```diff
@@ -183,36 +183,62 @@ In a Cloudflow application, Streamlets offer several customization options such
...
 ----
+cloudflow.streamlets.<streamlet-name>.<config-facet>.<config-key>=value
+----
+
+=== Supported Configuration Facets
 
-Note that because this file is in HOCON format, dot-notation is also supported:
+In the sandbox runtime, only two configuration facets of streamlets are currently supported:
+
+- configuration parameters, under the `config-parameters` facet.
```
Comment (RayRoestenburg) [inline docs/shared-content-source/docs/modules/develop/pages/cloudflow-local-sandbox.adoc:225]: facet -> section?

File: docs/shared-content-source/docs/modules/develop/pages/cloudflow-local-sandbox.adoc [hunk 2]
```diff
@@ -223,37 +249,37 @@ include::{cloudflow-examples-version}@docsnippets:ROOT:example$sensor-data-scala
...
 ----
-metric-logger {
-  msg-prefix = “local”
-}
+include::{cloudflow-examples-version}@docsnippets:ROOT:example$sensor-data-scala/src/main/resources/local-alt.conf[tag=config-param]
 ----
 
 === Using Volume Mounts
-In the Streamlet API, Volume Mounts are declared in a similar way to configuration parameters, using a programmatic description of the Volume Mount that includes its desired mount path. In the following example, we declare a Volume Mount with name source-data-mount, requested to be mounted at /mnt/data, and it requires to have a ReadWriteMany access mode:
+In the Streamlet API, volume mounts --a Kubernets construct to access disk storage-- are declared in a similar way to configuration parameters, using a programmatic description of the volume mount that includes its desired mount path.
```
Comment (yuchaoran2011) [inline docs/shared-content-source/docs/modules/develop/pages/cloudflow-local-sandbox.adoc:261]:
```suggestion
In the Streamlet API, volume mounts--a Kubernetes construct to access disk storage--are declared in a similar way to configuration parameters, using a programmatic description of the volume mount that includes its desired mount path.
```

File: docs/shared-content-source/docs/modules/develop/pages/cloudflow-local-sandbox.adoc [hunk 3]
```diff
@@ -223,37 +249,37 @@ include::{cloudflow-examples-version}@docsnippets:ROOT:example$sensor-data-scala
...
+In the Streamlet API, volume mounts --a Kubernets construct to access disk storage-- are declared in a similar way to configuration parameters, using a programmatic description of the volume mount that includes its desired mount path. 
+In the following example, we declare a volume mount with name `source-data-mount`, requested to be mounted at `/mnt/data` in the pod's filesystem, and it requires to have a `ReadWriteMany` access mode:
 
 [source,scala]
 ----
 include::{cloudflow-examples-version}@docsnippets:ROOT:example$sensor-data-scala/src/main/scala/sensordata/SensorDataFileIngress.scala[tag=volume-mount1]
 ----
 
-In a Kubernetes deployment, that Volume Mount gets mapped to a Permanent Volume Claim. The requested mount path is replaced by a local path when we use the Sandbox to run an application containing one or more streamlets that declare a Volume Mount. The local path assignment can be configured to point to a specific directory. Otherwise, the mount path will be assigned to a temporary directory, created on the fly.
+In a Kubernetes deployment, that volume mount gets mapped to a permanent volume claim (PVC).
```
Comment (yuchaoran2011) [inline docs/shared-content-source/docs/modules/develop/pages/cloudflow-local-sandbox.adoc:269]:
```suggestion
In a Kubernetes deployment, that volume mount gets mapped to a PersistentVolumeClaim (PVC). 
```

File: docs/shared-content-source/docs/modules/develop/pages/cloudflow-local-sandbox.adoc (hunk 1 above)
Comment (maasg) [reply docs/shared-content-source/docs/modules/develop/pages/cloudflow-local-sandbox.adoc:225]: yeah. I've been looking for a good term for that.
wdyt about _scope_?

Comment (RayRoestenburg) [reply docs/shared-content-source/docs/modules/develop/pages/cloudflow-local-sandbox.adoc:225]: hocon docs (once!) refers to config sections. I think config section is fine for now.

Comment (RayRoestenburg) [reply docs/shared-content-source/docs/modules/develop/pages/cloudflow-local-sandbox.adoc:225]: scope is good too.

File: docs/shared-content-source/docs/modules/develop/pages/cloudflow-local-sandbox.adoc [hunk 4]
```diff
@@ -183,36 +183,62 @@ In a Cloudflow application, Streamlets offer several customization options such
...
 
-rotor-avg-logger {
-  //config-key = value  -- values for the rotor-avg-logger streamlet
-}
+[source, HOCON]
 ----
+cloudflow.streamlets.<streamlet-name>.<config-facet>.<config-key>=value
+----
+
+=== Supported Configuration Facets
```
Comment (RayRoestenburg) [inline docs/shared-content-source/docs/modules/develop/pages/cloudflow-local-sandbox.adoc:221]:
```suggestion
=== Supported Configuration Scopes
```

File: docs/shared-content-source/docs/modules/develop/pages/cloudflow-local-sandbox.adoc [hunk 5]
```diff
@@ -183,36 +183,62 @@ In a Cloudflow application, Streamlets offer several customization options such
...
-  //config-key = value  -- values for the file-ingress streamlet
-}
+Note that because this file is in HOCON format, dot-notation is also supported.
 
-rotor-avg-logger {
-  //config-key = value  -- values for the rotor-avg-logger streamlet
-}
+[source, HOCON]
 ----
+cloudflow.streamlets.<streamlet-name>.<config-facet>.<config-key>=value
```
Comment (RayRoestenburg) [inline docs/shared-content-source/docs/modules/develop/pages/cloudflow-local-sandbox.adoc:218]:
```suggestion
cloudflow.streamlets.<streamlet-name>.<config-scope>.<config-key>=value
```

File: docs/shared-content-source/docs/modules/develop/pages/cloudflow-local-sandbox.adoc [hunk 6]
```diff
@@ -183,36 +183,62 @@ In a Cloudflow application, Streamlets offer several customization options such
...
+----
+
+=== Supported Configuration Scopes
 
-Note that because this file is in HOCON format, dot-notation is also supported:
+In the sandbox runtime, only two configuration facets of streamlets are currently supported:
```
Comment (RayRoestenburg) [inline docs/shared-content-source/docs/modules/develop/pages/cloudflow-local-sandbox.adoc:223]:
```suggestion
In the sandbox runtime, only two configuration scopes of streamlets are currently supported:
```

File: docs/shared-content-source/docs/modules/develop/pages/cloudflow-local-sandbox.adoc [hunk 7]
```diff
@@ -183,36 +183,62 @@ In a Cloudflow application, Streamlets offer several customization options such
...
+
+- configuration parameters, under the `config-parameters` facet.
```
Comment (RayRoestenburg) [inline docs/shared-content-source/docs/modules/develop/pages/cloudflow-local-sandbox.adoc:225]:
```suggestion
- configuration parameters, under the `config-parameters` scope.
```

File: docs/shared-content-source/docs/modules/develop/pages/cloudflow-local-sandbox.adoc [hunk 8]
```diff
@@ -183,36 +183,62 @@ In a Cloudflow application, Streamlets offer several customization options such
...
+- volume mounts, under the `volume-mounts` facet.
```
Comment (RayRoestenburg) [inline docs/shared-content-source/docs/modules/develop/pages/cloudflow-local-sandbox.adoc:226]:
```suggestion
- volume mounts, under the `volume-mounts` scope.
```

File: docs/shared-content-source/docs/modules/develop/pages/cloudflow-local-sandbox.adoc [hunk 9]
```diff
@@ -183,36 +183,62 @@ In a Cloudflow application, Streamlets offer several customization options such
...
+
+The general syntax for a configuration entry is:
+
+[source, HOCON]
 ----
-file-ingress.<config-key>=value
-rotor-avg-logger.<config-key>=value
+cloudflow.streamlets.valid-logger.<config-facet>.<config-key>=value
```
Comment (RayRoestenburg) [inline docs/shared-content-source/docs/modules/develop/pages/cloudflow-local-sandbox.adoc:232]:
```suggestion
cloudflow.streamlets.valid-logger.<config-scope>.<config-key>=value
```

File: docs/shared-content-source/docs/modules/develop/pages/cloudflow-local-sandbox.adoc [hunk 10]
```diff
@@ -183,36 +183,62 @@ In a Cloudflow application, Streamlets offer several customization options such
...
 ----
+where the `config-facet` may be either `config-parameters` or `volume-mounts`.
```
Comment (RayRoestenburg) [inline docs/shared-content-source/docs/modules/develop/pages/cloudflow-local-sandbox.adoc:234]:
```suggestion
where the `config-scope` may be either `config-parameters` or `volume-mounts`.
```

File: docs/shared-content-source/docs/modules/develop/pages/cloudflow-local-sandbox.adoc [hunk 11]
```diff
@@ -183,36 +183,62 @@ In a Cloudflow application, Streamlets offer several customization options such
...
+
+We can provide values for these two facets as we explain in the following section.
```
Comment (RayRoestenburg) [inline docs/shared-content-source/docs/modules/develop/pages/cloudflow-local-sandbox.adoc:236]:
```suggestion
We can provide values for these two scopes as we explain in the following section.
```

**Pull request merged**